asm_to_pil/src/lib.rs

```diff
@@ -352,32 +263,24 @@ impl<'a, T: FieldElement> ASMPILConverter<'a, T> {
             Register {
                 conditioned_updates,
                 default_update,
-                is_assignment: flag == Some(RegisterFlag::IsAssignment),
+                ty,
             },
         );
         self.pil.push(witness_column(start, name, None));
     }
 
     fn handle_instruction_def(
         &mut self,
-        InstructionDefinitionStatement {
-            start,
-            body,
-            name,
-            params,
-        }: InstructionDefinitionStatement<T>,
-    ) -> Option<Link<T>> {
-        let instruction_flag = format!("instr_{name}");
-        let instr = Instr {
-            name: name.clone(),
-            params: params.clone(),
-            flag: instruction_flag.clone(),
-        };
-        self.create_witness_fixed_pair(start, &instruction_flag);
+        s: InstructionDefinitionStatement<T>,
+    ) -> Option<InstructionDefinitionStatement<T>> {
+        let instruction_name = s.name.clone();
+        let instruction_flag = format!("instr_{instruction_name}");
```

Comment:
same here

Reply:
Agreed. This compiler steps assumes the existence of `instr_return` and `instr__reset`, which relies on the existence of `return` and `reset`. I do not have an immediate solution to that, apart from grouping all compiler steps together, which is not ideal. I'll have more of a think about it.

Reply:
Decided to make `return` first class citizen in functions and `return` and `reset` first class citizens in rom.

Reply:
Implemented only the former for now